django/extractor/service.py

```diff
@@ -41,12 +41,19 @@ def broadcast_events(
             event["record"] = record
             producer.produce_event(topic=f"{conf.PRODUCED_TOPIC_PREFIX}{batch_id}", event=event)
             count += 1
-    except BatchCancelled as err:
-        logger.warning({"message": str(err), "resource_id": resource_id, "batch_id": batch_id})
-        # return early to avoid setting the counter in redis
-        return
+    except KafkaException as err:
+        if err.args[0].code() == KafkaError.UNKNOWN_TOPIC_OR_PART:
+            logger.warning(
+                {"message": "The current batch has been cancelled", "resource_id": resource_id, "batch_id": batch_id}
+            )
+            # return early to avoid setting the counter in redis
+            return
+        else:
+            logger.exception(err)
     except EmptyResult as err:
         logger.warning({"message": str(err), "resource_id": resource_id, "batch_id": batch_id})
+    except ValueError as err:
+        logger.exception(err)
```

Comment:
since we're exiting the loop here, shouldn't we stop the batch at this point ?

Reply:
Oh you're right! But I don't think we want to stop the whole batch when we fail to produce an event in the extractor, I think the `try/except` should be inside the loop